src/components/molecules/AutoCompleteCheckbox/AutoCompleteCheckbox.tsx

```diff
@@ -0,0 +1,110 @@
+import { FC, useState } from 'react'
+import { Box, Button, Checkbox, Flex, Input, Popover, PopoverBody, PopoverContent, PopoverHeader, PopoverTrigger, Spacer, Stack, StackDivider, VStack } from '@chakra-ui/react'
+import { useTranslation } from 'react-i18next'
+
+import { SearchIcon } from '@/components/atoms/SearchIcon/SearchIcon'
+import { DropDownIcon } from '@/components/atoms/DropDownIcon/DropDownIcon'
+import { FilterCountIndicator } from '@/components/atoms/FilterCountIndicator/FilterCountIndicator'
+
+interface AutoCompleteCheckboxProps {
+  label: string
+  noOfSelectedFilters: number
+  options: string[]
+  selectedFilters: string[]
+  applyFilters: (values: string[]) => void
+}
+
+const AutoCompleteCheckbox: FC<AutoCompleteCheckboxProps> = (props) => {
+  const { label, options, applyFilters, noOfSelectedFilters, selectedFilters } = props
+  const [selectedValues, setSelectedValues] = useState<string[]>([])
+  const [searchInput, setSearchInput] = useState<string>(``)
+  const { t } = useTranslation(`search`)
+
+  const allChecked = options.length != 0 && selectedValues.length === options.length
+  const isIndeterminate = selectedValues.length !== 0 && selectedValues.length !== options.length
+
+  const onClose = () => {
+    setSearchInput(``)
+    setSelectedValues([...selectedFilters])
+  }
+
+  const onInputChange = (value: string) => {
+    setSearchInput(value)
+  }
+
+  const topCheckboxOnChange = () => {
+    if (isIndeterminate || selectedValues.length === 0) {
+      setSelectedValues([...options])
+    } else {
+      setSelectedValues([])
+    }
+  }
+
+  const checkboxOnChange = (checked: boolean, value: string) => {
+    if (checked) {
+      setSelectedValues([...selectedValues, value])
+    } else {
+      setSelectedValues((prevState) => prevState.filter((selectedValue) => selectedValue !== value))
+    }
+  }
```

Comment:
event handler functions should start with `handle`

Reply:
have refactored the functions to start with `handle`